source/utils/ethUtil.ts

```diff
@@ -78,18 +78,19 @@ export const estimateGasLimit = async ({ to, data }: { to: string, data: string
     
     const from = ethAsset.address;
     const abi = await _getAbi({to});
+    const web3 = _getWeb3(network);
+    const bytecode = await web3.eth.getCode(to);
 
+    // bytecode === '0x' is used for rinkeby testnet
     // Not a contract -> 21,000 standard gasLimit
-    if (!abi) {
+    if (!abi || bytecode === '0x') {
```

Comment:
Is there anything else we can do here to validate if this is an ETH transfer vs a smart contract call? This is where we fail if the ABI isn't published to Etherscan because 21,000 is way too low for any smart contract functions.

Reply:
I still think this should be on the implementer's side, transactions sent through the provider should contain a gas (gasLimit) field when sent, if not we should set 0x15f90 (90000) as the EIP says...
https://eips.ethereum.org/EIPS/eip-1474#eth_sendtransaction ...

It's the same reason why the `eth_estimateGas` should always be implemented when implementing `eth_sendTransaction` ...

The implementer knows which contract he is trying to connect the user to, I have not checked the MetaMask implementation but I may think they're not fetching ABIs from etherscan and based on that sending the transaction ...

Reply:
Another option would be to estimate gas ourselves. Directly with the transaction data using [`eth_estimateGas`](https://eips.ethereum.org/EIPS/eip-1474#eth_estimategas) ...

Reply:
I think for the 2nd, [this](https://github.com/StardustCollective/stargazer-wallet-ext/pull/328/files/8729c3f19d9f42815efb6e9889e0a907d61e5fda#r898284838) should be enough, the implementer must always send the `data` and `to` fields, with those 2 fields IMO I think we can estimate gas nicely ...